src/api/app/controllers/status_controller.rb

```diff
@@ -67,18 +67,11 @@ def workerstatus
 
   def history
     required_parameters :hours, :key
-    samples = begin
-      Integer(params[:samples] || '100') rescue 0
-    end
-    @samples = [samples, 1].max
 
-    hours = begin
-      Integer(params[:hours] || '24') rescue 24
-    end
-    starttime = Time.now.to_i - hours.to_i * 3600
-    # rubocop:disable Metrics/LineLength
-    @values = StatusHistory.where("time >= ? AND \`key\` = ?", starttime, params[:key]).pluck(:time, :value).collect { |time, value| [time.to_i, value.to_f] }
-    # rubocop:enable Metrics/LineLength
+    @samples = [params[:samples].to_i, 1].max
+    hours = params[:hours].to_i
```

Comment:
This could be replaced by using params[:hours] directly. The to_i call can be skipped. It's done in history_by_key_and_hours anyway.